pkgs/servers/x11/xorg/generate-expr-from-tarballs.pl

```diff
@@ -73,8 +74,12 @@
         next;
     }
 
+    # split by last occuring minus
+    my ($name, $version) = split(/-([^-]+)$/, $pkgName);
```

Comment:
Looks pretty good. I went and looked at how Nix does this:

https://github.com/NixOS/nix/blob/7c90552879da4d1df99b50c85e94201981e60123/src/libstore/names.cc#L22-L38

This catches cases where the version contains a hyphen, like `foo-bar-baz-1.0-rc3`. We could match this behavior with a slightly more advanced regex:

```
^(?P<name>.*?)(-(?P<version>[0-9].*))?$
```

this makes the version start with the first - separated section which starts with a digit. However, Nix's behavior is not strictly always correct either. For example, Nix would consider `font-adobe-utopia-75dpi-1.0.4`'s version to be `75dpi-1.0.4`. 

My opinion in this case is it is in these edge cases it is advantageous to be on the side of to having _more_ information in the `version` field and less in the `name` field, but that is just my very low stakes opinion.

Reply:
i updated the regex to match the last minus with a following number

Reply:
but wait a second. this messes things up

Reply:
i will use your version